src/core/queryClient.ts

```diff
@@ -111,6 +111,19 @@ export class QueryClient {
     return this.queryCache.find<TData>(queryKey, filters)?.state.data
   }
 
+  getQueriesData<TData = unknown>(queryKey: QueryKey): [QueryKey, TData][]
+  getQueriesData<TData = unknown>(filters: QueryFilters): [QueryKey, TData][]
+  getQueriesData<TData = unknown>(
+    queryKeyOrFilters: QueryKey | QueryFilters
+  ): [QueryKey, TData][] {
+    return this.getQueryCache()
+      .findAll(queryKeyOrFilters)
+      .map(({ queryKey, state }) => {
+        const data = state.data as TData
```

Comment:
I think the type assertion is fine. Users don't need to provide the `TData` generic, so they will get back `unknown` per default. Which is correct if you don't know the structure of the queries you are getting. For example, if you try to find all active queries, the structure of the result might be totally different, so `unknown` is good. If you, as a developer, know that you are only getting back todo-lists from `getQueriesData`, then you can basically do the assertion by adding the generic:

```
queryClient.getQueriesData<Array<Todo>>(['todo', 'list'])
```

so I think it just makes life easier for consumers 👍

Reply:
Thanks! Sounds good. I put a note in the docs about this caveat, but let me know if it isn't necessary.